core/src/test/java/org/hibernate/ogm/utils/TestableGridDialect.java

```diff
@@ -27,6 +28,13 @@
 	/**
 	 * Returns the number of entities in the datastore
 	 *
+	 * @param session
+	 */
+	long getNumberOfEntities(Session session);
```

Comment:
Some tests call this method before and after a rollback to make sure that count is correct.
It seems reasonable to execute this inside the same transaction.